lib/optimization/table_data_merge_columns_test.go

```diff
@@ -14,24 +14,24 @@ func TestTableData_UpdateInMemoryColumnsFromDestination_Tz(t *testing.T) {
 	{
 		// In memory and destination columns are both timestamp_tz
 		tableData := &TableData{inMemoryColumns: &columns.Columns{}}
-		tableData.AddInMemoryCol(columns.NewColumn("foo", typing.NewKindDetailsFromTemplate(typing.ETime, ext.TimestampTzKindType)))
+		tableData.AddInMemoryCol(columns.NewColumn("foo", typing.NewExtendedTimeDetails(typing.ETime, ext.TimestampTZKindType, "")))
 
-		assert.NoError(t, tableData.MergeColumnsFromDestination(columns.NewColumn("foo", typing.NewKindDetailsFromTemplate(typing.ETime, ext.TimestampTzKindType))))
+		assert.NoError(t, tableData.MergeColumnsFromDestination(columns.NewColumn("foo", typing.NewExtendedTimeDetails(typing.ETime, ext.TimestampTZKindType, ""))))
 		updatedColumn, isOk := tableData.inMemoryColumns.GetColumn("foo")
 		assert.True(t, isOk)
-		assert.Equal(t, ext.TimestampTzKindType, updatedColumn.KindDetails.ExtendedTimeDetails.Type)
+		assert.Equal(t, ext.TimestampTZKindType, updatedColumn.KindDetails.ExtendedTimeDetails.Type)
 	}
 	{
 		// In memory is timestamp_ntz and destination is timestamp_tz
 		tableData := &TableData{inMemoryColumns: &columns.Columns{}}
-		kd := typing.NewKindDetailsFromTemplate(typing.ETime, ext.TimestampNTZKindType)
+		kd := typing.NewExtendedTimeDetails(typing.ETime, ext.TimestampNTZKindType, "")
 		kd.ExtendedTimeDetails.Format = ext.RFC3339MillisecondNoTZ
```

Comment:
This can be passed into the constructor on the line above, right?

Reply:
Yes it can!